src/services/catalog.service.ts

```diff
@@ -70,6 +70,30 @@ export class CatalogService { static $inject = ['$filter', '$q', 'Constants', 'A
     return deferred.promise;
   }
 
+  // Get service plans for a service class, takes either the serviceClass object or it's name
+  public getServicePlansForServiceClass(serviceClass: any) : angular.IPromise < any > {
+    let plansVersion = this.apiService.getPreferredVersion('clusterserviceplans');
+    var opts: any = undefined;
+    let serviceClassName = _.get(serviceClass, 'metadata.name', serviceClass);
```

Comment:
I'd prefer to test `_.isString` explicitly so we don't accidentally handle values like `{}`

Reply:
Maybe

```ts
let serviceClassName = _.isString(serviceClass) ? serviceClass : _.get(serviceClass, 'metadata.name');
```